src/http_client/mod.rs

```diff
@@ -73,6 +75,25 @@ impl Body {
     }
 }
 
+impl Stream for Body {
+    type Item = io::Result<Bytes>;
+
+    fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        let mut buf = [0; 1];
+        match self.poll_read(cx, &mut buf) {
+            Poll::Ready(Ok(n)) => {
+                if n == 0 {
+                    Poll::Ready(None)
+                } else {
+                    Poll::Ready(Some(Ok(Bytes::from(&buf[..]))))
+                }
+            }
+            Poll::Pending => Poll::Pending,
+            Poll::Ready(Err(e)) => Poll::Ready(Some(Err(e))),
+        }
+    }
+}
+
```

Comment:
Ah, so we should probably use the `AsyncRead` variant in `async-compression`. This might be related to #28 so the body can be read without an extra `memcpy`, but I *definitely* do not want to expose `bytes` in any public API.

Reply:
I think it is, I don't think we will need bytes if `AsyncBufRead` is used.